turbo/jsonrpc/otterscan_search_v3.go

```diff
@@ -71,25 +66,23 @@ func (api *OtterscanAPIImpl) buildSearchResults(ctx context.Context, tx kv.Tempo
 			break
 		}
 
+		// Avoid reading the same block multiple times; multiple matches in the same block
+		// may be common.
+		mustReadBlock = mustReadBlock || blockNumChanged
+
 		// it is necessary to track dirty/lazy-must-read block headers
 		// because we skip system txs like rewards (which are not "real" txs
 		// for this rpc purposes)
-		mustReadHeader = mustReadHeader || blockNumChanged
 		if isFinalTxn {
 			continue
 		}
 
-		if mustReadHeader {
-			if header, err = api._blockReader.HeaderByNumber(ctx, tx, blockNum); err != nil {
```

Comment:
@JkLondon after switching api of `receiptsGenerator.GetReceipt` from block to header - will need to update user's code to read header instead of block (where can).